### ISSUE TITLE
MKS TFT32 cannot print from SD to MKS Gen-L running Marlin 1.1.9

### ISSUE THREAD

**Issue by AlwaysTinkering** [makerbase-mks/MKS-TFT#190]
I've got an MKS Gen-L v1.0 board running Marlin 1.1.9 (the latest version), and an MKS TFT32 board running the v3.02 firmware.  

Attempting to print a file from the SD card just hangs.  The timer continues to update, but it appears nothing is sent to the printer.  Sometimes it shows strange/clearly incorrect values for the temps when this happens, like 333/70 for the bed temp.  

The same file on the same SD card prints fine in the 12864 board that is also attached.  I was also able to update the MKS TFT firmware to 3.02 using this SD card, so I know the TFT-32 can read it.  It can also show and browse folders.  I have tried several other SD cards, and several different files, with the same result.  

Other TFT32 functions, like home, move, etc. all work fine and control the printer as expected.

After the printing hang, I can press the print cancel, and the printer will then home x and y.  After that, other functions work fine.  I have tried 512mb, 2gb, and 8gb SD cards, all with the same result.

**Comment by AlwaysTinkering** [comment]
I also tried printing from a USB memory stick.  This did not work either, hanging up exactly the same as the SD cards.

**Comment by AlwaysTinkering** [comment]
To confirm this is just an issue with the final stable release of Marlin, I took the MKS TFT28 that has been working fine with Marlin 1.1.8 on my older 3D printer, and hooked it to the new Marlin 1.1.9 printer.  It failed in the same way.  It went to the printing screen and sat there like it is waiting for something from the printer that Marlin no longer sends.  Perhaps the response sent by Marlin to some command has changed from what the TFT32 is expecting?  Since Marlin 1.1.9 is the final stable release, it seems like the version that should be supported by the TFT32.

**Comment by makerbase-mks** [comment]
Give me your configuration file.I need to test it.

**Comment by makerbase-mks** [comment]
I test the marlin 1.1.9 is ok.
Send your configuration to me. 
I need to test it.

**Comment by AlwaysTinkering** [comment]
I've attached my Marlin 1.1.9 config files as well as my MKS-TFT32 config file
[mks_config.txt](https://github.com/makerbase-mks/MKS-TFT/files/2386583/mks_config.txt)
[Marlin119Config.zip](https://github.com/makerbase-mks/MKS-TFT/files/2386582/Marlin119Config.zip)
I'm using the very latest August version of 1.1.9 from GitHub, not the earlier July version of 1.1.9.

**Comment by weed2all** [comment]
I think is because your mks config is under 250000 baud rate and your marlin is 115200...try to change either one to match other...

**Comment by weed2all** [comment]
[mks_config.txt](https://github.com/makerbase-mks/MKS-TFT/files/2407516/mks_config.txt)
Here you have the mks config file with the correct baud rate according to your marlin firmware...try it and let us know if it worked

**Comment by AlwaysTinkering** [comment]
[MarlinConfigsX5SBert.zip](https://github.com/makerbase-mks/MKS-TFT/files/2408922/MarlinConfigsX5SBert.zip)
Looks like I must have sent the wrong config, because in the config I'm using, they match.  If it were wrong, then the basic commands would not work either, and everything works fine except printing.  Also 250000 is the baud rate I'm using.  I think the config I sent was an earlier one from when I was using the melzi board that came with the printer, not the MKS Gen-L.  Regardless, attached is the config that I'm running.  
[MarlinConfigsX5SBert.zip](https://github.com/makerbase-mks/MKS-TFT/files/2408925/MarlinConfigsX5SBert.zip)

**Comment by AlwaysTinkering** [comment]
Just to verify, I reflashed the printer with the baud rate at 115200 and put a config on the TFT32 with the baud rate setting at 3 and rebooted to make sure it was applied.  Home, move, etc. work, just like before, but when attempting to print, it hangs up before even setting the bed and extruder temperatures.

**Comment by makerbase-mks** [comment]
The touch screen and the motherboard control are not normal. Generally, the probability of inconsistent baud rate is relatively large. Your current situation is, zero return, and movement are normal, but can't you heat and print right?
You can check if the led of the mos in front of the port will be lit when you warm up (it will be turned on when it is lit)
Or you would like to try to use the computer on the computer to control the motherboard, observe whether it can be heated normally, or print.

**Comment by AlwaysTinkering** [comment]
It is not a lack of 2-way communication.  I can go to the heat button and heat up the extruder, and the display correctly shows the temperature rise as it heats.  All functions work except print.  I can also print from Simplify3D and from an SD card in the 12864 Smart Graphic controller.  I can control and flash the printer from the computer.

**Comment by AlwaysTinkering** [comment]
Ok, I hooked up my logic analyzer between the TFT32 and the GEN-L.  Here is what's happening:

TFT32 sends:  N-1 M110*15
Marlin Replies:  ok N-1 P15 B3

(After 15 seconds)
TFT32 sends:  N-1 M110*15
Marlin Replies:  ok N-1 P15 B3

(After another 15 seconds)
TFT32 sends:  N-1 M110*15
Marlin Replies:  ok N-1 P15 B3

At this point, I stopped the capture.

It looks like some kind of handshaking issue with the handling of setting the line number.  The TFT doesn't seem to like the response.

The logic analyzer clearly shows the baud rate is correct on both sides, and that Marlin is sending a reply. The logic analyzer can read the reply, so I don't see why the TFT32 would not.  Like I said earlier, it can clearly read the temp from Marlin when setting the bed or extruder temps.  I think it just isn't recognizing the ok reply line with the planner and buffer space values included.

To verify my theory, I commented out the ADVANCED_OK define in Marlin's Configuration_adv.h, and reflashed the printer.  With ADVANCED_OK disabled, the TFT32 is able to print successfully with the printer.  Commenting out ADVANCED_OK turns off sending the Pn and Bn values in the ok reply.

**Comment by robustini** [comment]
The problem is not in Marlin but in the MKS-TFT firmware!

**Comment by srbaroc** [comment]
And well already know some solution?
After finishing the entire printer, including pid, step calibration etc ... I find myself with this problem, which with marlin 1.1.8 I did not have

**Comment by pjar82** [comment]
Yes I'm having this issue with the tft 28 and marlin 1.1.9 it shows the temp jumping from say 245 down to 233 and and up to 250 randomly has anyone fixed this issue yet

**Comment by manolocs** [comment]
Hi there, adding my problem to the discussion, I Have a Ender3+MKSGENL+TFT28, it was working perfect with manual bed leveling, I installed BLTouch and made some changes via Pronterface, when I select to print the temps rise but does not print at all, message printing on the display but nothing moving. any idea what can be wrong?

**Comment by RageFPV** [comment]
Bought a CR10S with the Creality V2.1 Board and an MKS TFT v3.0.3. Today I made the jump in updating whatever version of Marlin came on the printer to 1.1.9 following various guides etc. After autotuning and reuploading with new PID values etc I was ready to print. Selecting the file i wanted and confirming start the progress is shown with target bed and extruder temperatures. but the actual temps remained at 0 or whatever the last value was from Preheat/fillament loading. I could not get it to print.
I came accross this thread and now have it working thanks to AlwaysTinkering.
I needed to make a change in configuration_adv.h
Commented out ADVANCED_OK and now printer seems to be working. Bed heated up, everything homed, and extruder is heating after selecting a file to print. so as AlwaysTinkering suggested, there seems to be a feedback loop issue between the board and tft board when that ADVANCED_OK setting is enabled on marling 1.1.9?
Very confused by the whole marlin update process as my printer came with the touchscreen,  and all guides out there are focused on cr-10s with the standard LCD and swivel button.
Id like to work out how auto level and babysteps are done with the TFT.
